### PR TITLE
add flux-pmix

### PR DESCRIPTION
**Pull request by jan-janssen**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jan-janssen** [comment]
@conda-forge/staged-recipes, ready for review

**Comment by jan-janssen** [comment]
conda-forge/help-c-cpp

File: recipes/flux-pmix/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/flux-framework/flux-pmix/releases/download/v{{ version }}/flux-pmix-{{ version }}.tar.gz
+  sha256: f382800b1a342df0268146ea7ce33011299bf0c69a46ac8a52e87de6026c9322
+  patches:
+    - test.patch
```
Comment (ocefpaf) [inline recipes/flux-pmix/meta.yaml:12]: Can you add a comment on why this patch is needed?

Comment (jan-janssen) [reply recipes/flux-pmix/meta.yaml:12]: The tests cover the communication between multiple compute nodes. But on Azure we only have a single node available, so I disabled all the multi node tests.

Comment (ocefpaf) [reply recipes/flux-pmix/meta.yaml]: I meant add a comment in the recipe so we remember why we patched in the future. Right on top of the patch would be ok.

File: recipes/flux-pmix/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/flux-framework/flux-pmix/releases/download/v{{ version }}/flux-pmix-{{ version }}.tar.gz
+  sha256: f382800b1a342df0268146ea7ce33011299bf0c69a46ac8a52e87de6026c9322
+  patches:
+    - test.patch  # disable tests which require multiple compute nodes
```
Comment (jan-janssen) [inline recipes/flux-pmix/meta.yaml:12]: @ocefpaf I did this here

File: recipes/flux-pmix/meta.yaml (hunk 1 above)
Comment (jan-janssen) [reply recipes/flux-pmix/meta.yaml]: That is what I did.

**Pull request merged**